lib/mauth/client/security_token_cacher.rb

```diff
@@ -20,7 +13,7 @@ def initialize(mauth_client)
         end
 
         def get(app_uuid)
-          if !@cache[app_uuid] || @cache[app_uuid].expired?
+          if !@cache[app_uuid]
```

Comment:
can we get rid of this entire `@cache`, `@cache_write_lock`, and all the complicated logic to me?

the difference seems to be that this logic is only caching 200 responses in these instance vars, but would also caching 404s really be a problem? also, does mauth service even send cache headers on 404 responses?

Reply:
I can check

Reply:
caching 404 sounds dangerous to me. If the mauth server due to a bug returns 404 once, during the period we cache, the application will refuse any incoming request from that service. 
Didn't we have a big issue with caching errors from Eureka responses a few months back?

Reply:
i think that was something with 500 errors causing caches to be flushed.  idk that i'd be too concerned about mauth erroneously sending 404s, but you make a valid point.  however, it's moot because (after looking at the code from yohei's link above) it seems that it will not send any cache headers on 404 responses.  when token is not found an error gets raised on this line:

https://github.com/mdsol/mauth/blob/release.2019.2.1/app/controllers/security_tokens_controller.rb#L13

before the call to `expires_in` that sets the headers.  then the response is sent as a simple `head :not_found`:

https://github.com/mdsol/mauth/blob/release.2019.2.1/app/controllers/application_controller.rb#L67

and i don't think that sets any cache headers?

Reply:
It does not send any headers currently

Reply:
I could probably remove all, all of `@cache` should be going away, like @ejinotti-mdsol suggested.
but for now, all test pass, so we can test this and refactor another shot on top

Reply:
I agree, let's keep this PR minimal